wallet/src/de/schildbach/wallet/ui/main/HeaderBalanceFragment.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.main;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.AlphaAnimation;
+import android.view.animation.Animation;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+
+import org.bitcoinj.core.Coin;
+import org.bitcoinj.utils.Fiat;
+import org.dash.wallet.common.data.ExchangeRate;
+import org.dash.wallet.common.ui.CurrencyTextView;
+import org.dash.wallet.common.util.GenericUtils;
+
+import dagger.hilt.android.AndroidEntryPoint;
+import de.schildbach.wallet.Constants;
+import de.schildbach.wallet_test.R;
+
+@AndroidEntryPoint
+public final class HeaderBalanceFragment extends Fragment {
+    private View showBalanceButton;
```

Comment:
This file will need to be "merged" with the java of `dashpay` manually when the time comes.

Reply:
Or maybe not.  It depends on how the move is tracked with git.  In this PR, it is showed as a delete and create, rather than a move.

Reply:
I have thought about translating this file to Kotlin since it has become quite light after refactoring but decided it might be too many changes already for this PR.

I can handle the next merge if needed. This kind of refactoring that's done with `HeaderBalanceFragment` is what we can also do with `MainActivity` to significantly simplify merge in the future.